django_components/templatetags/component_tags.py

```diff
@@ -143,90 +166,149 @@ def render(self, context):
         # Thread safety: storing the context as a property of the cloned SlotNode without using
         # the render_context facility should be thread-safe, since each cloned_node
         # is only used for a single render.
-        cloned_node = SlotNode(self.name, self.nodelist)
-        cloned_node.parent_component = self.parent_component
-        cloned_node.context = context
-
-        with context.update({"slot": cloned_node}):
-            return self.get_nodelist(context).render(context)
+        # cloned_node = SlotNode(self.name, self.nodelist)
+        # cloned_node.parent_component = self.parent_component
+        # cloned_node.context = context
+        # with context.update({"slot": cloned_node}):
+        #     return self.get_nodelist(context).render(context)
 
-    def get_nodelist(self, context):
-        if ACTIVE_SLOT_CONTEXT_KEY not in context:
+        if FILLED_SLOTS_CONTEXT_KEY not in context:
             raise TemplateSyntaxError(
-                f"Attempted to render SlotNode {self.name} outside of a parent Component or "
-                "without access to context provided by its parent Component. This will not"
-                "work properly."
+                f"Attempted to render SlotNode '{self.name}' outside a parent component."
             )
-
-        overriding_nodelist = context[ACTIVE_SLOT_CONTEXT_KEY].get(
+        fill_node: Optional[FillNode] = context[FILLED_SLOTS_CONTEXT_KEY].get(
             self.name, None
         )
-        return (
-            overriding_nodelist
-            if overriding_nodelist is not None
-            else self.nodelist
-        )
-
-    def super(self):
-        """Render default slot content."""
-        return mark_safe(self.nodelist.render(self.context))
+        aliased_slot_var: Optional[UserSlotVar] = None
+        if fill_node is None:
+            nodelist = self.nodelist
+        else:
+            nodelist = fill_node.nodelist
+            if fill_node.alias is not None:
+                aliased_slot_var = UserSlotVar(self, context)
+        extra_context = {}
+        if aliased_slot_var:
+            extra_context[fill_node.alias] = aliased_slot_var
+        with context.update(extra_context):
+            return nodelist.render(context)
+
+    # def super(self):
+    #     """Render default slot content."""
+    #     return mark_safe(self.nodelist.render(self.context))
 
 
 @register.tag("slot")
 def do_slot(parser, token):
     bits = token.split_contents()
-    if len(bits) != 2:
-        raise TemplateSyntaxError("'%s' tag takes only one argument" % bits[0])
+    args = bits[1:]
+    # e.g. {% slot <name> %}
+    if len(args) == 1:
```

Comment:
In practice len(bits) will never be zero because, in order for the 'do_fill' tag function to be invoked, the 'slot' token must be encountered by the parser. In other words, len(bits) > 1 is guaranteed in all cases except when calling the tag function manually with token="" – which no-one is doing